tests/test_p2p_daemon.py

```diff
@@ -0,0 +1,343 @@
+import asyncio
+import multiprocessing as mp
+import subprocess
+from functools import partial
+from typing import List
+
+from hivemind.p2p.p2p_daemon_bindings.datastructures import ID
+
+import numpy as np
+import pytest
+
+from hivemind.p2p import P2P
+from hivemind.proto import dht_pb2
+
+RUNNING = 'running'
+NOT_RUNNING = 'not running'
+CHECK_PID_CMD = '''
+if ps -p {0} > /dev/null;
+then
+    echo "{1}"
+else
+    echo "{2}"
+fi
+'''
+
+
+def is_process_running(pid: int) -> bool:
+    cmd = CHECK_PID_CMD.format(pid, RUNNING, NOT_RUNNING)
+    return subprocess.check_output(cmd, shell=True).decode('utf-8').strip() == RUNNING
```

Comment:
In Bash, `if` just checks whether the command's return code is zero, so you can just call `ps -p {pid}` here and check its return status (without using the constant with the Bash code).